lib/internal/process/permission.js

```diff
@@ -0,0 +1,54 @@
+'use strict';
+
+const {
+  ObjectFreeze,
+  ArrayPrototypePush,
+} = primordials;
+
+const permission = internalBinding('permission');
+const { validateString, validateArray } = require('internal/validators');
+const { isAbsolute, resolve } = require('path');
+
+let experimentalPermission;
+
+module.exports = ObjectFreeze({
+  __proto__: null,
+  isEnabled() {
+    if (experimentalPermission === undefined) {
+      const { getOptionValue } = require('internal/options');
+      experimentalPermission = getOptionValue('--experimental-permission');
+    }
+    return experimentalPermission;
+  },
+  deny(scope, references) {
+    validateString(scope, 'scope');
+    if (references == null) {
+      return permission.deny(scope, references);
+    }
+
+    validateArray(references, 'references');
+    // TODO(rafaelgss): change to call fs_permission.resolve when available
+    const normalizedParams = [];
+    for (let i = 0; i < references.length; ++i) {
+      if (isAbsolute(references[i])) {
+        ArrayPrototypePush(normalizedParams, references[i]);
+      } else {
+        ArrayPrototypePush(normalizedParams, resolve(references[i]));
+      }
+    }
+
+    return permission.deny(scope, normalizedParams);
+  },
+
+  has(scope, reference) {
+    validateString(scope, 'scope');
+    if (reference != null) {
+      validateString(reference, 'reference');
```

Comment:
```suggestion
      // TODO: add support for WHATWG URLs and Uint8Arrays.
      validateString(reference, 'reference');
```

Reply:
I'll include it, but I don't get entirely why this is needed. Can you please explain?

Reply:
All Node.js APIs to point to files accept more than strings, for consistency this API should too.